src/EnergyPlus/RefrigeratedCase.cc

```diff
@@ -8794,8 +8795,18 @@ void SetupReportInput(EnergyPlusData &state)
                                         OutputProcessor::TimeStepType::Zone,
                                         OutputProcessor::StoreType::Sum,
                                         rack.Name);
+
+                    // if Location is Zone, GetRefrigerationInputEither checks that you have at least one load and that either:
+                    // * You have only cases, and they must be all in the same zone
+                    // * Or you must have a Heat Rejection Zone provided
+                    int rackZoneNum = -1;
+                    if (rack.HeatRejectionZoneNum > 0) {
+                        rackZoneNum = rack.HeatRejectionZoneNum;
+                    } else {
+                        rackZoneNum = RefrigCase(rack.CaseNum(1)).ActualZoneNum;
+                    }
                     SetupZoneInternalGain(state,
-                                          RefrigCase(rack.CaseNum(1)).ActualZoneNum,
+                                          rackZoneNum,
```

Comment:
When CoilFlag is not set, NumCoils == 0, and either you have NumWalkIns > 0, in which case you MUST have a heat rejection zone. Otherwise that means NumCases > 0 and it's enforced that all of them must be in the same zone.
(It also enforces that NumCoils + NumWalkIns + NumCases > 0, so we know there is a load)